core/src/main/scala/org/http4s/QueryParam.scala

```diff
@@ -196,6 +204,20 @@ object QueryParamDecoder {
         value.value.validNel
     }
 
+  def instantQueryParamDecoder(formatter: DateTimeFormatter): QueryParamDecoder[Instant] =
+    new QueryParamDecoder[Instant] {
+      override def decode(value: QueryParameterValue): ValidatedNel[ParseFailure, Instant] =
+        Either
```

Comment:
`Validated` has `catchOnly` and is a `Bifunctor`.  I don't think you need the `Either`.

Reply:
Done - https://github.com/http4s/http4s/pull/2724/commits/924c996094f74d947046b9b7ea12178df0fe0852